lib/apiservers/engine/backends/container.go

```diff
@@ -1070,13 +1092,47 @@ func (c *Container) Containers(config *types.ContainerListOptions) ([]*types.Con
 		}
 
 		// get the docker friendly status
-		_, status := dockerStatus(
+		exitCode, status := dockerStatus(
 			int(t.ProcessConfig.ExitCode),
 			t.ProcessConfig.Status,
 			t.ContainerConfig.State,
 			started,
 			stopped)
 
+		// labels func requires a config be passed
+		// TODO: refactor labelsFromAnnotations func for broader use
+		tempConfig := &containertypes.Config{}
+		err = labelsFromAnnotations(tempConfig, t.ContainerConfig.Annotations)
+		if err != nil && config.Filter.Include("label") {
+			return nil, fmt.Errorf("unable to convert vic annotations to docker labels (%s)", t.ContainerConfig.ContainerID)
+		}
+
+		listContext.Labels = tempConfig.Labels
+		listContext.ExitCode = exitCode
+		listContext.ID = t.ContainerConfig.ContainerID
+
+		// prior to further conversion lets determine if this container
+		// is needed or if the list is complete -- if the container is
+		// needed conversion will continue and the container will be added to the
+		// return array
+		action := filter.IncludeContainer(listContext, t)
+		switch action {
+		case filter.Exclude:
+			// skip to next container
+			continue payloadLoop
+		case filter.Stop:
+			// we're done
+			break payloadLoop
+		}
+
+		cmd := strings.Join(t.ProcessConfig.ExecArgs, " ")
+		// the docker client expects the friendly name to be prefixed
+		// with a forward slash -- create a new slice and add here
+		names := make([]string, 0, len(t.ContainerConfig.Names))
+		for i := range t.ContainerConfig.Names {
```

Comment:
I probably phrased it in a not clear way. clientFriendlyContainerName is a function, you create a copy here when you call this function. I also do not see any reason to take value directly from loop instead of access it each time by index.